sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -498,21 +473,15 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
       throw new TypeError("The parameter 'onError' must be of type 'function'.");
     }
 
-    this._createMessageSessionIfDoesntExist()
-      .then(async () => {
-        if (!this._messageSession) {
-          return;
-        }
-        if (!this._isClosed) {
-          this._messageSession.receive(onMessage, onError, options);
-        } else {
-          await this._messageSession.close();
-        }
-        return;
-      })
-      .catch((err) => {
+    if (!this._isClosed) {
+      try {
+        this._messageSession.receive(onMessage, onError, options);
+      } catch (err) {
         onError(err);
-      });
+      }
+    } else {
+      this._messageSession.close().catch((err) => onError(err));
```

Comment:
If `this._isClosed` is set doesn't that mean we're already in the process of closing this session? Do we need to do it here too?